engine/table/src/main/java/io/deephaven/engine/util/PyCallableWrapperJpyImpl.java

```diff
@@ -159,12 +183,17 @@ public void parseSignature() {
         }
 
         this.paramTypes = paramTypes;
+
+        returnType = (Class<?>) pyUdfDecoratedCallable.getAttribute("return_type", null);
```

Comment:
Because the `_py_udf` decorator returns the precise type in the case of return type is annotated as an array or other container types (list, tuple), not simply `PyObject`. As we are asked to support nD arrays (Avi), this is the direction we should go.  In the cases of return type being primitive, the two decorators get the same return type.

After this change, the `signature` field isn't relied upon for finding the return type but is still used for other purposes.